BackofficeBundle/Tests/Manager/RouteDocumentManagerTest.php

```diff
@@ -165,33 +165,29 @@ public function testCreateForNodeWithNoPublishedNode()
      * @param string $siteId
      * @param string $language
      *
-     * @dataProvider provideClearNodeData
+     * @dataProvider provideNodeData
      */
-    public function testClearForNode($nodeId, $siteId, $language)
+    public function testClearForNode($language, $id, array $aliasIds, $pattern, $exceptedPattern, $parentId = null)
     {
-        $node = Phake::mock('OpenOrchestra\ModelInterface\Model\NodeInterface');
-        Phake::when($node)->getLanguage()->thenReturn($language);
-        Phake::when($node)->getSiteId()->thenReturn($siteId);
-        Phake::when($node)->getNodeId()->thenReturn($nodeId);
 
-        $routeDocuments = new ArrayCollection();
-        Phake::when($this->routeDocumentRepository)->findByNodeIdSiteIdAndLanguage(Phake::anyParameters())->thenReturn($routeDocuments);
+        $nodeId = 'nodeId';
+        $childrenId = 'childrenId';
+        $node = $this->generateNode($language, $id, $pattern, $parentId, $nodeId);
+        $children = Phake::mock('OpenOrchestra\ModelInterface\Model\NodeInterface');
+        Phake::when($children)->getNodeId()->thenReturn($childrenId);
 
-        $routes = $this->manager->clearForNode($node);
+        Phake::when($this->nodeRepository)->findLastVersionByType($nodeId, $language, 'siteId')->thenReturn(array($node));
+        Phake::when($this->nodeRepository)->findByParent($nodeId, 'siteId')->thenReturn(array($children));
+        Phake::when($this->nodeRepository)->findByParent($childrenId, 'siteId')->thenReturn(array());
 
-        $this->assertSame($routeDocuments, $routes);
-        Phake::verify($this->routeDocumentRepository)->findByNodeIdSiteIdAndLanguage($nodeId, $siteId, $language);
-    }
+        $route = Phake::mock('OpenOrchestra\ModelInterface\Model\RouteDocumentInterface');
+        $routes = array($route);
+        Phake::when($this->routeDocumentRepository)->findByNodeIdSiteIdAndLanguage(Phake::anyParameters())->thenReturn($routes);
```

Comment:
I would suggest that you add a verification on this method call (on for the node parent, one for the node children)